internal/impl/mongodb/processor.go

```diff
@@ -315,9 +386,16 @@ func (m *Processor) ProcessBatch(ctx context.Context, spans []*tracing.Span, bat
 				Update: docJSON,
 				Hint:   hintJSON,
 			}
-		case client.OperationFindOne:
+		case client.OperationFindOne, client.OperationFindAndUpdate:
 			var decoded any
-			err := collection.FindOne(context.Background(), filterJSON, findOptions).Decode(&decoded)
+			var err error
+
+			if m.operation == client.OperationFindOne {
+				err = collection.FindOne(context.Background(), filterJSON, findOptions).Decode(&decoded)
+			} else {
+				err = collection.FindOneAndUpdate(context.Background(), filterJSON, docJSON, findAndUpdateOptions).Decode(&decoded)
```

Comment:
Hmm... Looking at this code makes me wonder if we should be using the `ctx` parameter from `ProcessBatch()` here instead of `context.Background()`. Same below for `collection.BulkWrite()`. This might be a bug...

Reply:
Exactly mt thoughts at first, but since other commands were using the background too, I used the same...

Best way to test this is by executing a long-execution command and then signal a shutdown?

Reply:
I think it might be hard to repro since Benthos calls `os.Exit` eventually. But yeah, it’s rude to let it terminate the operation that way.

Reply:
Fixed.